### DISCUSSION TITLE
Web API Template Documentation

### DISCUSSION THREAD

**Discussion by biapar** [Dotnet-Boxed/Templates#633] [Q&A]
Is there a documentation because you made something into your template web api?

**Comment by VictorioBerra** [comment]
I am not sure what your are asking. If you are asking for documentation for the Web API Template there is not much documentation needed, its all out on https://docs.microsoft.com/en-us/aspnet/core/?view=aspnetcore-3.1

You can find docs for the `dotnet new api` flags here: https://github.com/Dotnet-Boxed/Templates/blob/master/Docs/API.md

**Comment by biapar** [comment]
Hi, I mean of your code into the template. I installed and try to understand it.
How to use cache and other features that you put in.
PS: Do you not save by default a log file?

**Comment by VictorioBerra** [comment]
Most of the code inside the template is just regular ASPNET Core stuff. There is SOME boilerplate for a CRUD app in the following folders:

- `/Commands`
- `/Controllers`
- `/Repositories`

But these are pretty self explanatory IMO.

Check the .csproj file to see what packages we use that are not from Microsoft. Like [Swashbuckle.AspNetCore](https://github.com/domaindrivendev/Swashbuckle.AspNetCore) you will find the relevant docs on their site.

Now I will say, there is some stuff used in the `Startup.cs` that is from this project: https://github.com/Dotnet-Boxed/Framework Like anytime you see the `.If(...` thats some boxed stuff.

lastly, for the log question, no log file is generated by default but [Serilog is configured in our project by default to log to console and debug.](https://github.com/Dotnet-Boxed/Templates/blob/master/Source/ApiTemplate/Source/ApiTemplate/Program.cs#L134). If you want a log file, just add this config block to your appsettings or appsettings.Dev inside the Serilog key add this block:

```json
"WriteTo": [
    {
    "Name": "File",
        "Args": {
        "path": "%TEMP%\\Logs\\serilog-configuration-sample.txt",
        "outputTemplate": "{Timestamp:o} [{Level:u3}] ({Application}/{MachineName}/{ThreadId}) {Message}{NewLine}{Exception}"
        }
    }
]
```

**Comment by VictorioBerra** [comment]
I saw you edited your comment, if you want to know how to cache you need to visit the core docs.


For example, for **response cache**, the template does this: https://github.com/Dotnet-Boxed/Templates/blob/master/Source/ApiTemplate/Source/ApiTemplate/Startup.cs#L57
To see how that works, see docs here https://docs.microsoft.com/en-us/aspnet/core/performance/caching/response?view=aspnetcore-3.1

@biapar I think maybe your confusion is that this template is doing some custom stuff that is not out of the box in ASPNET Core. Almost every single thing we do in this template is just turning on stuff inside ASPNET Core that already exists. We also configure all those things with defaults that are more secure, or practical. Nothing we do here is magic. If you see something you dont understand, I highly recommend you just google it or consult the core docs.

**Comment by biapar** [comment]
Ok. I understand now. You build this template with .Net Core out of the box features but already configured and secured. My next step will be to put AD authentication with Jwt and use Vue.js as client side.
